http/server/api.go

```diff
@@ -18,28 +16,19 @@ import (
 )
 
 type rulesetAPI struct {
-	rulesets     api.RulesetService
-	timeout      time.Duration
-	watchTimeout time.Duration
+	rulesets api.RulesetService
 }
 
 func (s *rulesetAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	path := strings.TrimPrefix(r.URL.Path, "/rulesets/")
-
-	if _, ok := r.URL.Query()["watch"]; ok && r.Method == "GET" {
-		ctx, cancel := context.WithTimeout(r.Context(), s.watchTimeout)
-		defer cancel()
-		s.watch(w, r.WithContext(ctx), path)
-		return
-	}
-
-	ctx, cancel := context.WithTimeout(r.Context(), s.timeout)
-	defer cancel()
-	r = r.WithContext(ctx)
+	path := r.URL.Path
 
 	switch r.Method {
 	case "GET":
 		if _, ok := r.URL.Query()["list"]; ok {
```

Comment:
Isn't it more conventional to use r.Form ? That way the values are parsed only once FWIW.